docs/csharp/language-reference/builtin-types/reference-types.md

```diff
@@ -0,0 +1,178 @@
+
+```csharp
+string a = "hello";
+string b = "h";
+// Append to contents of 'b'
+b += "ello";
+Console.WriteLine(a == b);
+Console.WriteLine((object)a == (object)b);
+```
+
+This displays "true" and then "false" because the content of the strings are equivalent, but `a` and `b` do not refer to the same string instance.
+
+The + operator concatenates strings:
+
+```csharp
+string a = "good " + "morning";
+```
+
+This creates a string object that contains "good morning".
+
+Strings are *immutable*--the contents of a string object cannot be changed after the object is created, although the syntax makes it appear as if you can do this. For example, when you write this code, the compiler actually creates a new string object to hold the new sequence of characters, and that new object is assigned to b. The string "h" is then eligible for garbage collection.
+
+```csharp
+string b = "h";
+b += "ello";
+```
+
+The `[]` [operator](../operators/member-access-operators#indexer-operator-) can be used for readonly access to individual characters of a `string`:
+
+```csharp
+string str = "test";
+char x = str[2];  // x = 's';
+```
+
+In similar fashion, the `[]` operator can also be used for iterating over each character in a `string`:
+
+```csharp
+string str = "test";
+
+for (int i = 0; i < str.Length; i++)
+{
+  Console.Write(str[i] + " ");
+}
+// Output: t e s t
+``` 
+
+String literals are of type `string` and can be written in two forms, quoted and `@`-quoted. Quoted string literals are enclosed in double quotation marks ("):
+
+```csharp
+"good morning"  // a string literal
+```
+
+String literals can contain any character literal. Escape sequences are included. The following example uses escape sequence `\\` for backslash, `\u0066` for the letter f, and `\n` for newline.
+
+```csharp
+string a = "\\\u0066\n";
+Console.WriteLine(a);
+```
+
+> [!NOTE]
+> The escape code `\udddd` (where `dddd` is a four-digit number) represents the Unicode character U+`dddd`. Eight-digit Unicode escape codes are also recognized: `\Udddddddd`.
+
+[Verbatim string literals](../tokens/verbatim.md) start with `@` and are also enclosed in double quotation marks. For example:
+
+```csharp
+@"good morning"  // a string literal
+```
+
+The advantage of verbatim strings is that escape sequences are *not* processed, which makes it easy to write, for example, a fully qualified Windows file name:
+
+```csharp
+@"c:\Docs\Source\a.txt"  // rather than "c:\\Docs\\Source\\a.txt"
+```
+
+To include a double quotation mark in an @-quoted string, double it:
+
+```csharp
+@"""Ahoy!"" cried the captain." // "Ahoy!" cried the captain.
+```
+
+## The delegate type
+
+The declaration of a delegate type is similar to a method signature. It has a return value and any number of parameters of any type:
+
+```csharp
+public delegate void TestDelegate(string message);
+public delegate int TestDelegate(MyType m, long num);
```

Comment:
Should `Action` and `Func` .NET types be somehow mentioned here? I guess, nowadays those types are mostly used instead of declaring custom delegate types.